### PR TITLE
Add CUDA 12 conda installation docs

### LINKED ISSUE #9045: Docs should be updated for CUDA 12
The [current installation docs](https://numba.pydata.org/numba-doc/latest/cuda/overview.html#software) only mention `cudatoolkit` should be installed. However, CUDA 12 changes the way packages are organized and Numba should now require `cuda-nvcc=12.x` and I believe `cuda-version=12.x` too. Particularly, without `cuda-nvcc=12.x` installed, one might encounter errors such as:

```python
Traceback (most recent call last):
  File "<string>", line 7, in <module>
  File "/opt/conda/envs/test/lib/python3.10/site-packages/numba/cuda/cudadrv/runtime.py", line 111, in get_version
    self.cudaRuntimeGetVersion(ctypes.byref(rtver))
  File "/opt/conda/envs/test/lib/python3.10/site-packages/numba/cuda/cudadrv/runtime.py", line 65, in __getattr__
    self._initialize()
  File "/opt/conda/envs/test/lib/python3.10/site-packages/numba/cuda/cudadrv/runtime.py", line 51, in _initialize
    self.lib = open_cudalib('cudart')
  File "/opt/conda/envs/test/lib/python3.10/site-packages/numba/cuda/cudadrv/libs.py", line 63, in open_cudalib
    path = get_cudalib(lib)
  File "/opt/conda/envs/test/lib/python3.10/site-packages/numba/cuda/cudadrv/libs.py", line 55, in get_cudalib
    libdir = get_cuda_paths()[dir_type].info
  File "/opt/conda/envs/test/lib/python3.10/site-packages/numba/cuda/cuda_paths.py", line 223, in get_cuda_paths
    'nvvm': _get_nvvm_path(),
  File "/opt/conda/envs/test/lib/python3.10/site-packages/numba/cuda/cuda_paths.py", line 201, in _get_nvvm_path
    candidates = find_lib('nvvm', path)
  File "/opt/conda/envs/test/lib/python3.10/site-packages/numba/misc/findlib.py", line 44, in find_lib
    return find_file(regex, libdir)
  File "/opt/conda/envs/test/lib/python3.10/site-packages/numba/misc/findlib.py", line 56, in find_file
    entries = os.listdir(ldir)
FileNotFoundError: [Errno 2] No such file or directory: '/opt/conda/envs/test/nvvm/lib64'
```

cc @jakirkham @gmarkall

### PR DESCRIPTION
**Pull request by bdice**
This PR adds documentation for CUDA 12 conda installation methods.

Closes #9045.

### CODE REVIEW

**Comment by bdice** [comment]
cc: @jakirkham @pentschev @gmarkall for review. @gmarkall and anyone else with permission, feel free to push directly to this branch.

**Review by pentschev — COMMENTED** [review]
LGTM, thanks @bdice !

File: docs/source/cuda/overview.rst [hunk 1]
```diff
@@ -60,9 +60,15 @@ Presently 11.2 is the minimum required toolkit version. An NVIDIA driver
...
+Conda users can install the CUDA Toolkit into a conda environment.
 
-   $ conda install cudatoolkit
+For CUDA 12, ``cuda-nvcc`` and ``cuda-nvrtc`` are required::
+
+    $ conda install -c conda-forge cuda-nvcc cuda-nvrtc "cuda-version>=12.0"
+
+For CUDA 11, ``cudatoolkit`` is required::
+
+    $ conda install -c conda-forge cudatoolkit "cuda-version=11.*"
```
Comment (jakirkham) [inline docs/source/cuda/overview.rst:71]: Maybe we can drop the `*` from this and similar lines

It is effectively implied. Also `*` sometimes confuses shells

```suggestion
    $ conda install -c conda-forge cudatoolkit "cuda-version=11"
```

**Review by jakirkham — COMMENTED** [review]
Thanks Bradley! 🙏

This is a nice improvement 🙂

Added a few suggestions below

**Comment by stuartarchibald** [comment]
Thanks for the patch @bdice and for the reviews @pentschev and @jakirkham. 

@gmarkall please could you take a look, Many thanks.

File: buildscripts/condarecipe.local/meta.yaml [hunk 2]
```diff
@@ -54,8 +54,8 @@ requirements:
     # Exclude 0.3.20 too
     # https://github.com/numba/numba/issues/8096
     - libopenblas >=0.3.18, !=0.3.20     # [arm64]
-    # CUDA 10.2 or later is required for CUDA support
-    - cudatoolkit >=10.2
+    # CUDA 11.0 or later is required for CUDA support
+    - cudatoolkit >=11.0
```
[the remark this answers is not in the document]
Comment (gmarkall) [reply buildscripts/condarecipe.local/meta.yaml:58]: Thanks for the suggestion - I think it should be 11.2, as Numba no longer works with the NVVM based on LLVM 3.4 that was in pre-11.2 toolkits
```suggestion
    - cuda-version >=11.2
    - cudatoolkit >=11.2
```

File: docs/source/user/installing.rst [hunk 3]
```diff
@@ -45,12 +45,17 @@ Note that Numba, like Anaconda, only supports PPC in 64-bit little-endian mode.
...
 
-    $ conda install cudatoolkit
+For CUDA 12, ``cuda-nvcc`` and ``cuda-nvrtc`` are required::
 
-You do not need to install the CUDA SDK from NVIDIA.
+    $ conda install -c conda-forge cuda-nvcc cuda-nvrtc "cuda-version>=12.0"
+
+For CUDA 11, ``cudatoolkit`` is required::
 
+    $ conda install -c conda-forge cudatoolkit "cuda-version=11.*"
```
Comment (gmarkall) [inline docs/source/user/installing.rst:56]: Similar to @jakirkham's suggestion above:
```suggestion
    $ conda install -c conda-forge cudatoolkit "cuda-version=11"
```

File: numba/cuda/cudadrv/nvvm.py [hunk 4]
```diff
@@ -416,10 +416,14 @@ def get_arch_option(major, minor):
 
 
 MISSING_LIBDEVICE_FILE_MSG = '''Missing libdevice file.
-Please ensure you have package cudatoolkit >= 11.0
-Install package by:
+Please ensure you have a CUDA Toolkit 11.0 or higher.
```
Comment (gmarkall) [inline numba/cuda/cudadrv/nvvm.py:419]:
```suggestion
Please ensure you have a CUDA Toolkit 11.2 or higher.
```

File: numba/cuda/cudadrv/nvvm.py [hunk 5]
```diff
@@ -416,10 +416,14 @@ def get_arch_option(major, minor):
...
+For CUDA 12, ``cuda-nvcc`` and ``cuda-nvrtc`` are required:
 
-    conda install cudatoolkit
+    $ conda install -c conda-forge cuda-nvcc cuda-nvrtc "cuda-version>=12.0"
+
+For CUDA 11, ``cudatoolkit`` is required:
+
+    $ conda install -c conda-forge cudatoolkit "cuda-version=11.*"
```
Comment (gmarkall) [inline numba/cuda/cudadrv/nvvm.py:426]: As above:
```suggestion
    $ conda install -c conda-forge cudatoolkit "cuda-version=11"
```

File: buildscripts/condarecipe.local/meta.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jakirkham) [reply buildscripts/condarecipe.local/meta.yaml]: Yeah was also wondering about this. We use [11.2 in conda-forge]( https://github.com/conda-forge/numba-feedstock/blob/aa6fdf29d00c437f7a112ed1650d56ef7ec247c0/recipe/meta.yaml#L56-L58 ) too

Was having trouble finding this in the docs though. Do we have the minimum written down somewhere?

File: docs/source/cuda/overview.rst [hunk 6]
```diff
@@ -60,9 +60,15 @@ Presently 11.2 is the minimum required toolkit version. An NVIDIA driver
...
+Conda users can install the CUDA Toolkit into a conda environment.
 
-   $ conda install cudatoolkit
+For CUDA 12, ``cuda-nvcc`` and ``cuda-nvrtc`` are required::
+
+    $ conda install -c conda-forge cuda-nvcc cuda-nvrtc "cuda-version>=12.0"
+
+For CUDA 11, ``cudatoolkit`` is required::
+
+    $ conda install -c conda-forge cudatoolkit "cuda-version=11"
```
Comment (jakirkham) [inline docs/source/cuda/overview.rst:71]: If we have an 11.2 minimum, would it make sense to use this?

```suggestion
    $ conda install -c conda-forge cudatoolkit "cuda-version>=11.2,<12"
```

File: docs/source/user/installing.rst [hunk 7]
```diff
@@ -45,12 +45,17 @@ Note that Numba, like Anaconda, only supports PPC in 64-bit little-endian mode.
...
 
-    $ conda install cudatoolkit
+For CUDA 12, ``cuda-nvcc`` and ``cuda-nvrtc`` are required::
 
-You do not need to install the CUDA SDK from NVIDIA.
+    $ conda install -c conda-forge cuda-nvcc cuda-nvrtc "cuda-version>=12.0"
+
+For CUDA 11, ``cudatoolkit`` is required::
 
+    $ conda install -c conda-forge cudatoolkit "cuda-version=11"
```
Comment (jakirkham) [inline docs/source/user/installing.rst:56]: We may want to update this as well depending on how we handle the above comment

**Comment by jakirkham** [comment]
Had some questions about incorporating the 11.2 minimum

Otherwise this looks good

**Comment by bdice** [comment]
@gmarkall @jakirkham Sorry, I realized I had left off a crucial piece. I just updated the Minor Version Compatibility docs.

I also updated the minimum to 11.2 everywhere I could see and added a reference to CUDA 12.4 now that it's available. If we need to split this PR somehow, let me know.

**Comment by jakirkham** [comment]
LGTM. Thanks Bradley! 🙏

File: docs/source/cuda/minor_version_compatibility.rst [hunk 8]
```diff
@@ -5,39 +5,61 @@ CUDA Minor Version Compatibility
...
+
+   conda install -c rapidsai pynvjitlink
+
+To install with pip, use the NVIDIA package index:
+
+.. code:: bash
+
+   pip install --extra-index-url https://pypi.nvidia.com pynvjitlink-cu12
+
+CUDA 12
```
Comment (gmarkall) [inline docs/source/cuda/minor_version_compatibility.rst:45]:
```suggestion
CUDA 11
```

**Review by gmarkall — APPROVED** [review]
Looks good to me! Just waiting for CI prior to marking ready to merge.

**Comment by bdice** [comment]
Looks like CI passed. @gmarkall This should be ready to merge.

**Pull request merged**